framework/resource/manager.go

```diff
@@ -348,6 +349,34 @@ func (m *Manager) DestroyAll(args ...interface{}) error {
 	// Call it
 	result := finalFunc.Call(mapperArgs...)
 
+	// Populate the declared/destroyed resources. The declared resources are the resources
+	// which remain after destroying, and the destroyed resources are the ones that have
+	// been destroyed (which implement WithDestroy). If a resource does not implement a
+	// destroy function, then it is a declaredResource. If it does, it's a destroyedResource
+	if m.dcr != nil || m.dtr != nil {
+		for name, resource := range m.resources {
+			if resource.destroyFunc != nil {
+				destroyedResource, err := resource.DestroyedResource()
+				if err != nil {
+					m.logger.Debug("Failed to destroy declared resource",
+						"resource name", name,
+						"platform", resource.platform,
+						"error", err,
+					)
+					return nil
+				}
+
+				m.dtr.DestroyedResources = append(m.dtr.DestroyedResources, destroyedResource)
```

Comment:
It looks like we are adding all destroyed resources to the `DestroyedResources` list even if they were not successfully destroyed whereas we raise an error for declared resources. Is that expected?